lib/termsupport.zsh

```diff
@@ -95,8 +95,10 @@ function omz_termsupport_preexec {
     fi
   fi
 
-  # cmd name only, or if this is sudo or ssh, the next cmd
-  local CMD="${1[(wr)^(*=*|sudo|ssh|mosh|rake|-*)]:gs/%/%%}"
+  # cmd name only, or if this is doas/sudo or ssh, the next cmd
+  local _subex
+  zstyle -s ':omz' 'subexecutor' _subex
+  local CMD="${1[(wr)^(*=*|${_subex}|_|subex|mosh|rake|-*)]:gs/%/%%}"
```

Comment:
I replaced `sudo` in the filter with `${_subex}|_|subex` for the following reasons:

* The subexecutor is not necessarily `sudo` now (can be `doas`, for instance)
* There's an underscore `_` alias (previously defined in `lib/misc.zsh`) that aliases the subexecutor
* There's a new global function `subex` that also invokes the subexecutor